packages/karma-runner/src/KarmaTestRunner.ts

```diff
@@ -32,27 +40,32 @@ export default class KarmaTestRunner implements TestRunner {
     this.setGlobals(setup, getLogger);
     this.cleanRun();
     this.listenToServerStart();
-    this.listenToRunComplete();
     this.listenToSpecComplete();
     this.listenToCoverage();
     this.listenToError();
   }
 
-  public init(): Promise<void> {
+  public async init(): Promise<void> {
     return new Promise((res, rej) => {
-      StrykerReporter.instance.once('browsers_ready', res);
-      this.starter
-        .start()
-        .then(() => {
-          /*noop*/
-        })
-        .catch(rej);
+      StrykerReporter.instance.once('browsers_ready', () => res());
+      this.starter.start().catch(rej);
     });
   }
 
-  public async run({ testHooks }: { testHooks?: string }): Promise<RunResult> {
-    this.testHooksMiddleware.currentTestHooks = testHooks || '';
-    if (this.currentRunStatus !== RunStatus.Error) {
+  public dryRun(options: DryRunOptions): Promise<DryRunResult> {
+    TestHooksMiddleware.instance.configureCoverageAnalysis(options.coverageAnalysis);
+    return this.run();
+  }
+
+  public async mutantRun(options: MutantRunOptions): Promise<MutantRunResult> {
+    TestHooksMiddleware.instance.configureActiveMutant(options);
+    const dryRunResult = await this.run();
+    return toMutantRunResult(dryRunResult);
+  }
+
+  private async run(): Promise<DryRunResult> {
+    // this.testHooksMiddleware.currentTestHooks = testHooks || '';
```

Comment:
This was the other instance of commented code that I just mentioned

Reply:
Thanks 😅